### PR TITLE
Implemented an initial version of domain DMN model's abstraction.

### PR DESCRIPTION
**Pull request by mariuszkumor**

### CODE REVIEW

File: src/main/java/org/powerflows/dmn/domain/model/input/InputType.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.powerflows.dmn.domain.model.input;
+
+public enum InputType {
+    STRING,
+    INTEGER,
+    BOOLEAN;
```
Comment (stlhrt) [inline src/main/java/org/powerflows/dmn/domain/model/input/InputType.java]: we should hande also other types like a collection

File: src/main/java/org/powerflows/dmn/domain/model/builder/AbstractBuilder.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    protected T product;
+
+    {
+        initProduct();
+    }
+
+    public T build() {
+        T product = assembleProduct();
+        initProduct();
```
Comment (stlhrt) [inline src/main/java/org/powerflows/dmn/domain/model/builder/AbstractBuilder.java]: did I say I don't like re usable builders?  still true...

File: src/main/java/org/powerflows/dmn/domain/model/output/OutputType.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.powerflows.dmn.domain.model.output;
+
+public enum OutputType {
```
Comment (stlhrt) [inline src/main/java/org/powerflows/dmn/domain/model/output/OutputType.java]: I thing this should be a ValueType and be common to both Input and Output

File: src/test/groovy/org/powerflows/dmn/domain/model/DecisionSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+        rule1InputEntry2Expression.getValue() == someRule1InputEntry2ExpressionValue
+
+        final OutputEntry rule1OutputEntry1 = rule1.getOutputEntries().get(0)
+        rule1OutputEntry1.getName() == someRule1OutputEntry1Name
+
+        final Expression rule1OutputEntry1Expression = rule1OutputEntry1.getExpression()
+        rule1OutputEntry1Expression.getType() == someRule1OutputEntry1ExpressionType
+        rule1OutputEntry1Expression.getValue() == someRule1OutputEntry1ExpressionValue
+
+        final Rule rule2 = decision.getRules().get(1)
```
Comment (stlhrt) [inline src/test/groovy/org/powerflows/dmn/domain/model/DecisionSpec.groovy]: please use a 
```groovy
with(decision.getRules().get(1)) {
   getDescription() == someRule2Description
}
```
and tbh I'd rather write
```groovy
with(decision.rules[1]) {
   description == someRule2Description
}
```

File: src/main/java/org/powerflows/dmn/domain/model/builder/AbstractBuilder.java (hunk 2 above)
Comment (mariuszkumor) [reply src/main/java/org/powerflows/dmn/domain/model/builder/AbstractBuilder.java]: Changed

File: src/test/groovy/org/powerflows/dmn/domain/model/DecisionSpec.groovy [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+        with(decision) {
+            getId() == someTableId
+            getName() == someTableName
+            getHitPolicy() == someHitPolicy
+            getInputs().size() == 2
+            getOutputs().size() == 2
+            getRules().size() == 2
+        }
+
+        final Input input1 = decision.getInputs().get(0)
```
Comment (stlhrt) [inline src/test/groovy/org/powerflows/dmn/domain/model/DecisionSpec.groovy]: this variable is used only once... the same applies to the ones below

Comment (mariuszkumor) [reply src/test/groovy/org/powerflows/dmn/domain/model/DecisionSpec.groovy]: In this way is more clean in my opinion.

**Pull request merged**